### ISSUE TITLE
How to save N ensemble members as a dimension ? (as ncecat does)

### ISSUE THREAD

**Issue by PBrockmann** [senesis/climaf#119]
Hi,

I would like to aggregate N ensemble members on a new dimension as I can do it with the `ncecat` operator.

I have tried this using a cscript that calls `ncecat` but the resulting file contains `ph_r1i1p1f1` and `ph_r2i1p1f1` (variable with suffixes) rather than an single `ph` variable with a dimension record set to 2.

<pre>
from climaf.api import *
#========================================
s1 = ds(project='CMIP6', model='*', variable='ph', experiment='historical', period='1850', frequency='monthly', realization='r[12]i1*', version='latest')
e1 = s1.explore('ensemble')
#========================================
from natsort import natsorted
e1.set_order(natsorted(e1.order))
#========================================
e1_surf = ccdo(e1, operator='sellevidx,1')
#========================================
e1_t = space_average(e1_surf)
#========================================
cscript('myncecat','ncecat ${ins} ${out}')
e1_m = myncecat(e1_t)
#========================================
efile(e1_m, '/data/brocksce/tyty.nc', forced=True)
</pre>

The interest to get all members on a unique dimension is strong since you can pass directly any operator on this new dimension (variance, max, min).

**Comment by senesis** [comment]
Yes, `efile` complains that `e1_m` is not an ensemble; this, because CliMAF objects of type ensemble up to now can be generated only by :
- `eds()`
- dataset's method `explore()` , with arg `'ensemble'` (as in your script)
- or applying a non-ensemble-capable operator to an ensemble

But when using an ensemble-capable operator (i.e. declared using ${mmin}, such as my_ncecat with my proposed change ), the (implicit) assumption is that the output is a single member dataset (and you can use `cfile()` on it)

I understand that you expected to handle the output of `ncecat` as a CliMAF ensemble, but this is not (yet) possible. Could you try to go as far as possible given this limitation and using this output as a single member dataset having the additional `record` dimension (`record` being the name chosen by ncecat for the dimension representing the members) ?

**Comment by PBrockmann** [comment]
Might be a good thing to let this issue open since the feature is not covered yet ?

**Comment by senesis** [comment]
Issue reopen, but needed changes may be quite tricky
And yes, pointing here from a FAQ entry would be useful

**Comment by senesis** [comment]
By the way, is it really necessary to handle e1_m as an ensemble for your purpose ? : I understand that you wish to apply on e1_m some operators that can handle this data organization (namely : a single file with a record dimension representing the member number). In that case, you can : 
- write e1_m using cfile()
- handle that file as a CliMAF dataset (and not an ensemble) using [function fds()](https://climaf.readthedocs.io/en/latest/functions_data.html?highlight=fds#fds-define-a-dataset-from-a-data-file)
- declare these operators using cscript() and apply them as if the data was not an ensemble. I assume that most of these operators usually output a single field rather than a member. But for the other ones, ultimately, if wishing to get back to CliMAF capabilities regarding handling ensemble, we could re-build an ensemble a call to cens() without args, followed  by a loop on extracting one index on dimension record and adding to the ensemble

Am I clear enough ? 
S
